parsons/newmode/newmode.py

```diff
@@ -323,3 +327,270 @@ def get_outreach(self, outreach_id, params={}):
+
+    def convert_to_table(self, data):
+        """Internal method to create a Parsons table from a data element."""
+        table = None
+        if type(data) is list:
+            table = Table(data)
+        else:
+            table = Table([data])
+
+        return table
+
+    def base_request(self, method, url, data=None, json=None, data_key=None, params={}):
+        """
+        Internal method to make a call to Newmode API and validate the response
+        """
+        response = None
+        response = self.client.request(
+            url=url, req_type=method, json=json, data=data, params=params
+        )
+        response.raise_for_status()
+        success_codes = [200, 201, 202, 204]
+        self.client.validate_response(response)
+        if response.status_code in success_codes:
+            if self.client.json_check(response):
+                response_json = response.json()
+                return response_json[data_key] if data_key else response_json
+            else:
+                return response.status_code
+        return response
+
+    def converted_request(
+        self,
+        endpoint,
+        method,
+        supports_version=True,
+        data=None,
+        json=None,
+        params={},
+        convert_to_table=True,
+        data_key=None,
+    ):
+        """Internal method to make a call to the Newmode API and convert the result to a Parsons table."""
+        self.client = OAuth2APIConnector(
+            uri=self.base_url,
+            auto_refresh_url=V2_API_AUTH_URL,
+            client_id=self.client_id,
+            client_secret=self.__client_secret,
+            headers=self.headers,
+            token_url=V2_API_AUTH_URL,
+            grant_type="client_credentials",
+        )
+
+        url = f"{self.api_version}/{endpoint}" if supports_version else endpoint
+        response = self.base_request(
+            method=method,
+            url=url,
+            json=json,
+            data=data,
+            params=params,
+            data_key=data_key,
+        )
+        if not response:
+            logging.warning(f"Empty result returned from endpoint: {endpoint}")
+        if convert_to_table:
+            return self.convert_to_table(response)
+        else:
+            return response
+
+    def get_campaign(self, campaign_id, params={}):
+        """
+        Retrieve a specific campaign by ID.
+
+        In v2, a campaign is equivalent to Tools or Actions in V1.
+        `Args:`
+            campaign_id: str
+                The ID of the campaign to retrieve.
+            params: dict
+                Query parameters to include in the request.
+        `Returns:`
+            Parsons Table containing campaign data.
+        """
+        endpoint = f"/campaign/{campaign_id}/form"
+        response = self.converted_request(
+            endpoint=endpoint,
+            method="GET",
+            params=params,
+        )
+        return response
```

Comment:
nit: by calling the variable "response", I assume this is the full response. I'd call it "data" or even just do 
```suggestion
        return self.converted_request(
            endpoint=endpoint,
            method="GET",
            params=params,
        )
```